### PR TITLE
fix(CtCompilationUnitImpl): toString() no longer throws a NPE

### LINKED ISSUE #2913: Cannot get imports of a file
Hi,
I'm trying to get the imports from a source file, but the implementation seems to be buggy.
Here's my code:
`        CompilationUnit cu = launcher.getFactory().CompilationUnit().getOrCreate(srcTojavaFile);
        ModelList<CtImport> imports = cu.getImports(); `

No matter what I put in the file, `cu.getImports()` always returns empty. Using the debugger, seems in the CompilationUnit object imports are always empty.

Thanks!
Luca

### PR DESCRIPTION
**Pull request by nharrand**
fix #2913

### CODE REVIEW

**Comment by monperrus** [comment]
Thanks. With a test case?

File: src/test/java/spoon/reflect/visitor/ImportScannerImplTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		CtType t = launcher.getModel().getRootPackage().getPackage("matchers").getType("TestClass");
+
+		CompilationUnit cu = launcher.getFactory().CompilationUnit().getOrCreate(t);
+		List<CtImport> imports = cu.getImports();
+
+		//contract: import are accessible from CompilationUnit
+		assertEquals(imports.size(), 2);
+
+		//contract: CompilationUnitImpl#toString() does not throw a NPE
+		assertNotNull(cu.toString());
```
Comment (monperrus) [inline src/test/java/spoon/reflect/visitor/ImportScannerImplTest.java]: assert on the content? `assertEquals`

File: src/test/java/spoon/reflect/visitor/ImportScannerImplTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		launcher.addInputResource("src/test/resources/simple-import/TestClass.java");
+		launcher.getEnvironment().setAutoImports(true);
+		launcher.buildModel();
+
+		CtType t = launcher.getModel().getRootPackage().getPackage("matchers").getType("TestClass");
+
+		CompilationUnit cu = launcher.getFactory().CompilationUnit().getOrCreate(t);
+		List<CtImport> imports = cu.getImports();
+
+		//contract: import are accessible from CompilationUnit
```
Comment (msteinbeck) [inline src/test/java/spoon/reflect/visitor/ImportScannerImplTest.java]: import -> imports

File: src/test/java/spoon/reflect/visitor/ImportScannerImplTest.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+		List<CtImport> imports = cu.getImports();
+
+		//contract: imports are accessible from CompilationUnit
+		assertEquals(imports.size(), 2);
+
+		//contract: CompilationUnitImpl#toString() does not throw a NPE
+		assertNotNull(cu.toString());
+
+		//contract: CompilationUnitImpl#toString() returns the file's name.
+		assertEquals(cu.toString(), "TestClass.java");
```
Comment (monperrus) [inline src/test/java/spoon/reflect/visitor/ImportScannerImplTest.java]: expected first thx

**Comment by monperrus** [comment]
Thanks!

**Pull request merged**